File: Day02/Part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	horizontal := 0
	vertical := 0
	aim := 0

	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer file.Close()

	//loop over the file and add each line to a slice
	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	//for each line in lines, split line and print the result
	for _, line := range lines {
		splitLine := strings.Split(line, " ")
		v, _ := strconv.Atoi(splitLine[1])
		switch splitLine[0] {
		case "forward":
			horizontal += v
			vertical += aim * v
		case "up":
			aim -= v
		case "down":
			aim += v
		}
	}

	fmt.Println("Distance: ", horizontal*vertical)

}

File: Day02/Part2.py
horizontal = 0
vertical = 0
aim = 0

with open('input.txt', 'r') as f:
    for line in f:
        ## split line by space and store in list
        line = line.split()
        if line[0] == "forward":
            horizontal += int(line[1])
            vertical += aim * int(line[1])
        elif line[0] == "down":
            aim += int(line[1])
        elif line[0] == "up":
            aim -= int(line[1])

    ##multiply horizontal and vertical to get total distance
    total = horizontal * vertical
    print(total)


        


